### PR TITLE
Add pill radio button

### PR DESCRIPTION
**Pull request by rickyrombo**
### Description

Adds a new component that's an accessible "Pill" radio button
image

Currently localized to the BuyAudoModal but can easily extract to stems if we need to use elsewhere.

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/mjp-pill-radio-button

File: packages/web/src/components/buy-audio-modal/components/PillRadioButton.module.css [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  display: none;
+}
+
+.labelText {
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+
+  /* height: 45px; */
```
Comment (dharit-tan) [inline packages/web/src/components/buy-audio-modal/components/PillRadioButton.module.css:11]: nit: remove comment?

**Review by dharit-tan — APPROVED** [review]

**Review by dylanjeffers — APPROVED** [review]
This looks great! so a few callouts:

1. Do we want to define a radio-group component?
2. Are there use cases where the pill is not a radio button and just a button?
3. Do we want :focus styles, like when someone tabs to the button?
4. Do we want to move this to stems at some point?

**Comment by rickyrombo** [comment]
> This looks great! a few callouts would love to hear your thoughts:
> 
> 1. Do we want to define a radio-group component?
> 2. Are there use cases where the pill is not a radio button and just a button?
> 3. Do we want :focus styles, like when someone tabs to the button?
> 4. Do we want to move this to stems at some point?
> 5. Do we want to have accessibilty options to have a better label, instead of just a number? so like it would read `5 audio`, `10 audio` etc? there are a few options here if we wanna go there

My take: yes to all of this!
1. On it!
2. This already exists in Stems
3. This sounds like a Julian question, but I think that would be good! I'll add to my list of Julian Q's
4. I can move it to stems now if you'd like. I don't see anywhere else this is used currently (the others are all buttons).
5. Would love to pick your brain here - would that not be simply specifying the `aria-label` on the `inputProps`?

**Comment by dylanjeffers** [comment]
> > This looks great! a few callouts would love to hear your thoughts:
> > 
> > 1. Do we want to define a radio-group component?
> > 2. Are there use cases where the pill is not a radio button and just a button?
> > 3. Do we want :focus styles, like when someone tabs to the button?
> > 4. Do we want to move this to stems at some point?
> > 5. Do we want to have accessibilty options to have a better label, instead of just a number? so like it would read `5 audio`, `10 audio` etc? there are a few options here if we wanna go there
> 
> My take: yes to all of this!
> 
> 1. On it!
> 2. This already exists in Stems
> 3. This sounds like a Julian question, but I think that would be good! I'll add to my list of Julian Q's
> 4. I can move it to stems now if you'd like. I don't see anywhere else this is used currently (the others are all buttons).
> 5. Would love to pick your brain here - would that not be simply specifying the `aria-label` on the `inputProps`?

1-3 sg
4. yeahh... since one already exists in stems maybe we hold off to see if this would be a general pattern in a few places?
5. yeah i think just passing aria-label could be good, since you already have a label element thought i think it is best to update that one. not sure what is absolute best way, but i think using a visually hidden span might be the move there

**Review by piazzatron — APPROVED** [review]

File: packages/stems/src/components/RadioButtonGroup/RadioButtonGroup.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    (e) => {
+      setValueState(e.target.value)
+      if (onChange) {
+        onChange(e)
+      }
+    },
+    [setValueState, onChange]
+  )
+  return (
+    <RadioGroupContext.Provider value={{ name, onChange: handleChange, value }}>
```
Comment (dylanjeffers) [inline packages/stems/src/components/RadioButtonGroup/RadioButtonGroup.tsx:44]: love that you added the context and everything

File: packages/stems/src/components/RadioButtonGroup/RadioButtonGroup.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  defaultValue?: string
+  value?: string | null
+} & ComponentPropsWithoutRef<'div'>
+
+export const RadioButtonGroup = (props: RadioButtonGroupProps) => {
+  const {
+    name,
+    onChange,
+    children,
+    value: valueProp,
```
Comment (dylanjeffers) [inline packages/stems/src/components/RadioButtonGroup/RadioButtonGroup.tsx:25]: also love that we are using the `valueProp` rename + the useControlled

File: packages/stems/src/components/RadioButtonGroup/RadioButtonGroup.tsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    defaultValue,
+    ...divProps
+  } = props
+  const [value, setValueState] = useControlled({
```
Comment (dylanjeffers) [inline packages/stems/src/components/RadioButtonGroup/RadioButtonGroup.tsx:29]: personally like spaces between large hook blocks :)

File: packages/stems/src/components/RadioPillButton/RadioPillButton.tsx [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  if (radioGroup) {
+    if (typeof name === 'undefined') {
+      name = radioGroup.name
+    }
+    if (typeof checked === 'undefined') {
+      checked = String(props.value) === String(radioGroup.value)
+    }
+  }
+
+  const handleChange = useCallback(
```
Comment (dylanjeffers) [inline packages/stems/src/components/RadioPillButton/RadioPillButton.tsx:42]: awesome stuff

**Review by dylanjeffers — APPROVED** [review]
looks absolutely perfect

**Pull request merged**